src/components/GlobalPreferences/Notifications/Notifications.js

```diff
@@ -1,24 +1,178 @@
+  const [email, setEmail] = useState(
+    localStorage.getItem(NOTIFICATION_SERVICE_EMAIL_KEY)
+  )
+  const [token, setToken] = useState(
+    localStorage.getItem(NOTIFICATION_SERVICE_TOKEN_KEY)
+  )
+
+  useEffect(() => {
+    if (!email && !token) {
+      setAuthState(AUTH_UNAUTHENTICATED)
+      return
+    }
+
+    if (email && !token) {
+      setAuthState(AUTH_PREVERIFY)
+      return
+    }
+
+    if (email && token) {
+      setAuthState(AUTH_AUTHENTICATING)
+      isAuthTokenValid(token)
+        .then(v => {
+          setAuthState(AUTH_AUTHENTICATED)
+          return v
+        })
+        .catch(e => {
+          setAuthState(AUTH_AUTHENTICATION_FAILED)
+        })
+    }
+  }, [email, token])
+
+  useEffect(() => {
+    token
+      ? localStorage.setItem(NOTIFICATION_SERVICE_TOKEN_KEY, token)
+      : localStorage.removeItem(NOTIFICATION_SERVICE_TOKEN_KEY)
+  }, [token])
+
+  useEffect(() => {
+    email
+      ? localStorage.setItem(NOTIFICATION_SERVICE_EMAIL_KEY, email)
+      : localStorage.removeItem(NOTIFICATION_SERVICE_EMAIL_KEY)
+  }, [email])
+
+  const handleLogout = useCallback(() => {
+    setEmail(null)
+    setToken(null)
+  }, [setToken, setEmail])
+
+  return {
+    authState,
+    email,
+    token,
+    handleTokenChange: setToken,
+    handleEmailChange: setEmail,
+    handleLogout,
+  }
+}
+
+export default function Notifications({
+  apps,
+  subsection,
+  dao,
+  handleNavigation,
+  navigationIndex,
+}) {
+  const navigateToNotifications = useCallback(
+    () => handleNavigation(navigationIndex),
+    [handleNavigation, navigationIndex]
   )
+  const {
+    authState,
+    email,
+    token,
+    handleTokenChange,
+    handleEmailChange,
+    handleLogout,
+  } = useAuthState()
+
+  if (subsection && subsection.startsWith(VERIFY_SUBSECTION)) {
+    return (
+      <NotificationsVerify
+        subsection={subsection}
+        onTokenChange={handleTokenChange}
+        onEmailChange={handleEmailChange}
+        navigateToNotifications={navigateToNotifications}
+      />
+    )
+  }
+
+  switch (authState) {
+    case AUTH_PREVERIFY:
+      return <NotificationsPreVerify email={email} />
+    case AUTH_AUTHENTICATED:
+      return (
+        <ManageNotifications
+          dao={dao}
+          onLogout={handleLogout}
+          apps={apps}
+          email={email}
+          token={token}
+        />
+      )
+    case AUTH_UNAUTHENTICATED:
+      return (
+        <NotificationsLogin
+          dao={dao}
+          authState={authState}
+          onEmailChange={handleEmailChange}
+        />
+      )
+    case AUTH_AUTHENTICATING:
+      return (
+        <NotificationsVerifyBox header="Authenticating">
+          <div
+            css={`
+              display: flex;
+              justify-content: center;
+              align-items: center;
+            `}
+          >
+            <LoadingRing />
+          </div>
+        </NotificationsVerifyBox>
+      )
+    case AUTH_AUTHENTICATION_FAILED:
```

Comment:
Same thing happening here with the containing `grid`:

![image](https://user-images.githubusercontent.com/3072458/63692129-3de61980-c811-11e9-9504-9320cc6bd835.png)